extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -346,16 +348,12 @@ export class AmpStoryPage extends AMP.BaseElement {
     this.markMediaElementsWithPreload_();
     this.initializeMediaPool_();
     this.maybeCreateAnimationManager_();
-    this.maybeSetPreviewDuration_();
-    this.advancement_ = AdvancementConfig.forElement(this.win, this.element);
-    this.advancement_.addPreviousListener(() => this.previous());
-    this.advancement_.addAdvanceListener(() =>
-      this.next(/* opt_isAutomaticAdvance */ true)
-    );
-    this.advancement_.addProgressListener((progress) =>
-      this.emitProgress_(progress)
-    );
```

Comment:
I don't think these lines should be moved inside the `onPreviewStateUpdate_()`, they don't really belong there.

Reply:
Just to add a final update before I leave of what was discussed offline: This use case is very specific, and it hides the setting/configuration of this.advacement_ to a place that may be not very obvious. This is not a good idea, since it is a very important part of the story page.

A possible solution would be to make a method that does that setup, and call it explicitly on the beginning, without using the initialization on the state listener. This way, our edge case stays out of the way of the normal setup